### ISSUE TITLE
[Bug][SFT] Validation uses drop_last=True and train_batch_size, which can drop the entire val set and log NaN val/loss

### ISSUE THREAD

**Issue by YeonwooSung** [verl-project/verl#7464]
### System Info

Reproduced by reading current main (`cfacd76a`).

Affects both SFT entry points:
- `verl/trainer/sft_trainer.py` (SPMD / torchrun)
- `verl/trainer/sft_trainer_ray.py` (Ray)

Default config: `verl/trainer/config/sft_trainer_engine.yaml`

```yaml
data.train_batch_size: 256
data.micro_batch_size_per_gpu: 4  # comment: "this is also val batch size"
```

### Information

- [x] The official example scripts
- [ ] My own modified scripts

### Tasks

- [x] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

Both SFT trainers build the **validation** loader with the **training** global batch and `drop_last=True` on the sampler **and** the dataloader:

```python
# verl/trainer/sft_trainer.py
self.train_batch_size_per_dp = self.global_batch_size // dp_size  # train_batch_size, default 256

self.val_sampler = DistributedSampler(
    self.val_dataset, shuffle=False, num_replicas=dp_size, rank=dp_rank, drop_last=True
)
self.val_dataloader = StatefulDataLoader(
    dataset=self.val_dataset,
    batch_size=self.train_batch_size_per_dp,
    ...
    drop_last=True,
)
```

`sft_trainer_ray.py` does the same (`dp_size=1`, so batch size is the full `train_batch_size`).

The config comment says `micro_batch_size_per_gpu: 4  # this is also val batch size`, but that field is never used for val.

Then validation always reduces whatever the loader yielded:

```python
val_losses = []
for val_data in self.val_dataloader:
    ...
    val_losses.append(metrics["loss"])

val_loss = torch.mean(torch.tensor(val_losses, device=self.device_name))
# logged as val/loss
```

`torch.mean(torch.tensor([]))` is `nan`. No warning. Exit code is 0.

PPO already does this correctly (`ray_trainer.py` / `v1/trainer_base.py`): `drop_last=False` and `val_batch_size` defaulting to `len(val_dataset)`.

Not a duplicate of #7401 (epoch-boundary resume skips the next train epoch) or #467 / #6319 (train-loss NaN on Qwen3.5).

#### Concrete numbers

| Setup | Val samples | Per-rank val samples | Val batch | Batches after `drop_last=True` |
|---|---:|---:|---:|---:|
| Ray SFT, default `train_batch_size=256`, 200-sample val | 200 | 200 | 256 | **0** → `val/loss = NaN` |
| SPMD SFT, 8 GPUs, `train_batch_size=256`, 100-sample val | 100 | 12 | 32 | **0** → `val/loss = NaN` |
| Same, 300-sample val | 300 | 37 | 32 | 1 (5 samples dropped) |

#### Minimal CPU repro of the empty-reduce

```python
import torch
from torch.utils.data import Dataset, DataLoader, DistributedSampler

class Toy(Dataset):
    def __len__(self):
        return 200
    def __getitem__(self, i):
        return i

# Ray SFT path: dp_size=1, batch_size=train_batch_size=256
loader = DataLoader(
    Toy(),
    batch_size=256,
    sampler=DistributedSampler(Toy(), num_replicas=1, rank=0, shuffle=False, drop_last=True),
    drop_last=True,
)
print("num val batches:", len(loader))          # 0
print("mean of empty:", torch.mean(torch.tensor([])).item())  # nan
```

### Expected behavior

If `val_files` is set, validation should score the full val set (or a dedicated `val_batch_size`) and never log NaN just because the tail does not fill a train-sized batch.

#### Proposed fix

Match the PPO val loader:

- `drop_last=False` on the val sampler and val dataloader.
- Val batch size = `data.val_batch_size` if set, else `data.micro_batch_size_per_gpu` (honour the existing comment), else `len(val_dataset)`.
- If `val_losses` is empty, log a warning and skip `val/loss` instead of reducing an empty tensor.

Apply in both `sft_trainer.py` and `sft_trainer_ray.py`.
